staging/src/k8s.io/apimachinery/pkg/util/net/port_range.go

```diff
@@ -59,19 +58,39 @@ func (pr *PortRange) Set(value string) error {
 	}
 
 	hyphenIndex := strings.Index(value, "-")
-	if hyphenIndex == -1 {
-		return fmt.Errorf("expected hyphen in port range")
+	plusIndex := strings.Index(value, "+")
+
+	// exclusive or
+	if (hyphenIndex != -1) == (plusIndex != -1) {
+		return fmt.Errorf("unable to parse port range: %s", value)
 	}
 
 	var err error
 	var low int
 	var high int
-	low, err = strconv.Atoi(value[:hyphenIndex])
-	if err == nil {
-		high, err = strconv.Atoi(value[hyphenIndex+1:])
+
+	if hyphenIndex != -1 {
+		low, err = strconv.Atoi(value[:hyphenIndex])
+		if err == nil {
```

Comment:
indent error flows https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

event though it's longer, this is more readable:
```golang
low, err = strconv.Atoi(value[:hyphenIndex])
if err == nil {
  ...
}
high, err = strconv.Atoi(value[hyphenIndex+1:])
if err != nil {
  ...
}
```